app/External/Slack/Modals/EquipmentAuthorization.php

```diff
@@ -96,26 +96,19 @@ public static function handle(SlackRequest $request)
         $selectedEquipment = self::equipmentFromState($state);
         $selectedMembers = self::peopleFromState($state);
         
-        $actorId = $request->customer()->id;
-
-        foreach($selectedMembers->crossJoin($selectedEquipment)->all() as [$person, $equipment]) {
-            if (!$person->hasMembership($equipment->user_plan_id)) {
-                Log::info('EquipmentAuthorization: Customer '.$actorId.' authorized Customer '.$person->id.' to use equipment under plan id '.$equipment->user_plan_id);
-                $api->members->addMembership($person->id, $equipment->user_plan_id);
-            }
-
-            if ($makeTrainers && !$person->hasMembership($equipment->trainer_plan_id)) {
-                Log::info('EquipmentAuthorization: Customer '.$actorId.' authorized Customer '.$person->id.' to train on equipment with plan id '.$equipment->trainer_plan_id);
-                $api->members->addMembership($person->id, $equipment->trainer_plan_id);
+        $actor = $request->customer();
+
+        try {
+            app()->make(BatchAuthorizeEquipmentAction::class)->execute($actor, $selectedMembers, $selectedEquipment, $makeTrainers);
+        } catch (\Exception $e) {
+            if ($e->getMessage() == 'NotAuthorized') {
+                return response()->json([
+                    'response_action' => 'errors',
+                    'errors' => [self::EQUIPMENT_DROPDOWN => "You don't have permission to authorize members for this equipment."]
+                ]);
             }
         }
 
-
-        if (! $makeTrainers && ! $makeUsers) {
-            return (new FailureModal('Neither user nor trainer appear to have been selected. Try again?'))
-                ->update();
-        }
-
```

Comment:
We should be able to have the `submit_disabled` set to True on the modal if the requirements aren't fulfilled. Maybe something to add as an enhancement?

Reply:
If you want to use `submit_disabled`, let's talk about it in an issue. The code that I've removed here does not set `submit_disabled`, so I'm not sure what that has to do with this change.

I removed this code block because removing `$makeUsers` means it's not longer possible for the form to "do nothing" when submitted with users and equipment are selected - which is what this failure mode was trying to communicate.

As for the `no users || no equipment` messages and disabling the submit: 
My preference for UX is for primary action buttons to never be disabled, and for form errors to be shown after submission. I'd rather allow the user to make a mistake and then point it out, than have a user be frustrated because the submit button doesn't work. 

Conveniently, that's exactly how it works right now. IMO our "no user selected" and "no equipment selected" are unnecessary/redundant; the fields are already required by default so submission with no values leads to an appropriate error message adjacent to the field: 

![Screenshot 2024-01-14 at 10 40 26 PM](https://github.com/Denhac/denhac-webhooks/assets/22403528/90e924da-fb46-4451-918b-d80425c7e0c6)

That's how it worked in the old UI, too:

![Screenshot 2024-01-14 at 11 04 43 PM](https://github.com/Denhac/denhac-webhooks/assets/22403528/dbe7c090-9e96-47fd-9b10-588f4c6422a5)

Reply:
Ah sweet, then yeah screw it let's just get rid of noEquipient and noPerson stuff then.

Reply:
Addressed in #52